### DISCUSSION TITLE
LinLibertineFont in OpenMage?

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2360] [Ideas]
This font can be found in **/lib/LinLibertineFon**t. There are new versions of it but they have not been updated in Magento/OpenMage. The font is used in the CAPTCHA images which are exceeded in my opinion because they are far from how they should be compared to Google reCAPTCHA. 

Those who still use Mage_Captcha do nothing but annoy their users. In the past, through a PR, the font was changed in the PDF invoices and switched to Helvetica. Recently I opened a discussion about Mage_Capcha future, whether we keep it and improve it a little or eliminate it. 

LinLibertineFont was not the best choice as it is a serif, a sans-serif would have been preferable for readability, but I don't know what was the reasoning behind it being chosen in the past. Maybe because it was free. Personally, I would remove it and if Mage_Captcha module is kept, replace it with another one.

**[OFFTOPIC]** an analysis of the directories in /lib should be done, which of them are still mandatory or simply relics. I understand that some directories will be removed from the repo and will be added through Composer. I hope that the situation in which Composer is not used has also been foreseen, how to add manually at least phpseclib/phpseclib and phpseclib/mcrypt_compat to OpenMage.

**Comment by kiatng** [comment]
I am actually using the font like like:
```php
    /**
     * Draw barcode, does not reset font
     *
     * @return $this
     */
    protected function _drawBarcode($pdf, $text, $pageIndex=0, $topOffset=765, $leftOffset=410)
    {
        $barcodeOptions = array(
            'text'    => $text,
            'font'    => Mage::getBaseDir() . '/lib/LinLibertineFont/LinLibertineC_Re-2.8.0.ttf',
            'factor'  => 1.55,
            'drawText'=> false // reduce file size
        );
        $rendererOptions = array(
            'topOffset'  => $topOffset,
            'leftOffset' => $leftOffset
        );
        Zend_Barcode::factory('code39', 'pdf', $barcodeOptions, $rendererOptions)->setResource($pdf, $pageIndex)->draw();
        return $this;
    }
```

**Reply by sreichel** [reply]
Pls check https://docs.laminas.dev/laminas-barcode/usage/

LinLibertineFont is part of Mage_Captcha an can/will maybe removed.
